nnunetv2/training/nnUNetTrainer/nnUNetLightningModule.py

```diff
@@ -0,0 +1,869 @@
+                                                                       if ignore_label is not None else regions,
+                                                                       'target', 'target'))
+
+        if deep_supervision_scales is not None:
+            tr_transforms.append(DownsampleSegForDSTransform2(deep_supervision_scales, 0, input_key='target',
+                                                              output_key='target'))
+        tr_transforms.append(NumpyToTensor(['data', 'target'], 'float'))
+        tr_transforms = Compose(tr_transforms)
+        return tr_transforms
+
+    @staticmethod
+    def get_validation_transforms(deep_supervision_scales: Union[List, Tuple],
+                                  is_cascaded: bool = False,
+                                  foreground_labels: Union[Tuple[int, ...], List[int]] = None,
+                                  regions: List[Union[List[int], Tuple[int, ...], int]] = None,
+                                  ignore_label: int = None) -> AbstractTransform:
+        val_transforms = []
+        val_transforms.append(RemoveLabelTransform(-1, 0))
+
+        if is_cascaded:
+            val_transforms.append(MoveSegAsOneHotToData(1, foreground_labels, 'seg', 'data'))
+
+        val_transforms.append(RenameTransform('seg', 'target', True))
+
+        if regions is not None:
+            # the ignore label must also be converted
+            val_transforms.append(ConvertSegmentationToRegionsTransform(list(regions) + [ignore_label]
+                                                                        if ignore_label is not None else regions,
+                                                                        'target', 'target'))
+
+        if deep_supervision_scales is not None:
+            val_transforms.append(DownsampleSegForDSTransform2(deep_supervision_scales, 0, input_key='target',
+                                                               output_key='target'))
+
+        val_transforms.append(NumpyToTensor(['data', 'target'], 'float'))
+        val_transforms = Compose(val_transforms)
+        return val_transforms    
+        
+    def configure_optimizers(self):
+        #TODO: This is just so that the checkpoint loading works using nnUNet functionality
+        # Would want to move to Lightning based checkpointing
+        if self.nnUNet_optimizer is None:
+            optimizer = torch.optim.SGD(self.model.parameters(), self.initial_lr, weight_decay=self.weight_decay,
+                                        momentum=0.99, nesterov=True)
+            lr_scheduler = PolyLRScheduler(optimizer, self.initial_lr, self.num_epochs)
+        else:
+            optimizer = self.nnUNet_optimizer
+            lr_scheduler = self.nnUNet_lr_scheduler
+        
+        return {"optimizer": optimizer, "lr_scheduler": lr_scheduler}
+    
+    def on_train_start(self):
+        # All of the relevant logic is already in `setup` 
```

Comment:
Fyi, setup, prepare_data, the splitting functionality, dataloaders, and all the transform things are generally recommended (but not required necessarily) to go into a separate `LightningDataModule`. So `on_train_start` pertains to trainer + model specific things. Just wanted to comment that the hooks do have separate intentions.

Reply:
Ok noted. I think in an immediate next PR we should plan to break out the Data loading related functionality into a `LightningDataModule` . My understanding was having a `LightningDataModule` is not necessary, so I avoided it to keep things as simple for the first step.

Reply:
Yeah it's not necessary---99% of things in the LightningDataModule is implemented in LightningModule, but if we ever want to process datasets different ways and be able to swap things, the separate data module is better.

Agreed on the PR for data loading.